### DISCUSSION TITLE
Zeek or Suricata to generate metadata?

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#7898] [Unsupported Versions] [answered]
In the setup there is a step that allows the choice of Zeek or Suricata to generate metadata. The default setting is for Zeek. Is there a reason or use case for selecting Suricata? Why would I choose one over the other?

**Comment by mnasec** [comment]
I had problems with Zeek packetloss, Suricata packetloss and Capture packetloss with 16cores and running zeek and suricata at 4Gbit/s Sniffing traffic. I tried to use the NUMA architecture and put the right cores for the suriprocs on the network interface, but it did not solve the problem in general. At this point i decided to switch off zeek and use the metadata from suricata. Packetloss is mostly gone now and most of the metadata is available, but some fields are not mapped correctly, like dns.answers.name. But i looked into the message field to find the result for the A record. Another topic are some missing x509 informations. 
So as a conclusion, if i had enough cores i would use zeek and suricata, but as a workaround suricata for metadata works also.

**Reply by dougburks** [reply]
> but some fields are not mapped correctly, like dns.answers.name

I suspect this is most likely because you switched an existing Zeek installation to Suricata metadata and the old Zeek data is still in Elasticsearch. This may resolve itself over time as the old Zeek indices are purged.

**Comment by dougburks** [comment, marked as answer]
@FredSenhenenmenen In general, both Zeek and Suricata do a great job of generating metadata. If possible, we recommend that you try one and then try the other so that you can see how each tool's metadata represents the traffic on your network.

**Reply by ghost** [reply]
When I use Zeek and look at Network in Kibana the protocol cloud just displays DHCP and DNS. When I use Suricata and look at Network in Kibana the cloud displays Failed and DHCP. This is a home lab and not a lot of traffic.

**Reply by dougburks** [reply]
Even in a home lab, I would expect to see more than just DHCP. You may be seeing only broadcast traffic. You might need to review how you're sniffing traffic from your network and consider a tap or span port if you're not already doing so:
https://docs.securityonion.net/en/2.3/hardware.html#packets

**Reply by ghost** [reply]
![Screenshot 2022-05-07 065323](https://user-images.githubusercontent.com/104913237/167251224-c8f6d96a-1e72-418a-9923-db51f1f3c3bf.png)
We are using the mirror port on our network switch connected to the network interface for monitoring traffic on the Security Onion.

**Reply by TotieBash** [reply]
I suggest use NTAP they are about $50-$75 on ebay.  
![76C25E8D-5685-4A68-B858-AA231CED9177](https://user-images.githubusercontent.com/86748904/168434982-9c2c3b92-113b-45be-93f6-2dba5fcf20cd.png)

As far as Zeek va Suricata for metadata I tested both and I really wanted to just use Suricata and turn off Zeek altogether but I keep coming back to it.  For my use case I use Zeek conn log to give me Netflow like data and Zeek gives me client.bytes and server.bytes that Suricata does not.  I have a modified SO connection kibana dashboard that I label Zeek Flow that gives me graph of top talkers.  In addition, just like what you already found out on SO builtin dashboards they are mostly built with Zeek metadata in mind.

Also, is this vm?

**Reply by ghost** [reply]
Thank you very much for this reply. This is very helpful. Since this is a lab environment, I can change all of this around and see what happens/works best. This is not a VM, it is a dedicated server. I have found the problem with the network sniffing: I just had to reboot the switch and the mirror port is working fine.

**Reply by dougburks** [reply]
> Zeek gives me client.bytes and server.bytes that Suricata does not.

To be clear, Suricata does provide `client.ip_bytes` and `server.ip_bytes`.

**Reply by TotieBash** [reply]
Thanks Doug I will look at it again, I looked at it last year and last month and dud not see it.  Since you said it I will reevaluate again because if that is the case I might use Suricata for metadata and not use Zeek at all.

**Comment by ghost** [comment]
It is working properly now. I rebooted the switch and everything is fine. Don't know why that was necessary, but it did the trick. Thanks for taking the time to help me with this.
